parser-typechecker/src/Unison/Codebase/MainTerm.hs

```diff
@@ -43,23 +44,30 @@ getMainTerm loadTypeOfTerm parseNames mainName mainType =
       let refs = NamesWithHistory.lookupHQTerm hq (NamesWithHistory.NamesWithHistory parseNames mempty)
       let a = Parser.Ann.External
       case toList refs of
+        [] -> pure (NotFound mainName)
         [Referent.Ref ref] -> do
           typ <- loadTypeOfTerm ref
           case typ of
             Just typ ->
-              if Typechecker.isSubtype typ mainType
+              if Typechecker.fitsScheme typ mainType
                 then do
                   let tm = DD.forceTerm a a (Term.ref a ref)
                   return (Success hq tm typ)
                 else pure (BadType mainName $ Just typ)
             _ -> pure (BadType mainName Nothing)
-        _ -> pure (NotFound mainName)
+        _ -> pure (error "multiple matching refs") -- TODO: make a real exception
```

Comment:
Not sure if you were planning to add the exception in this PR or not, but here's a reminder 😄